src/systems/systems/src/pressurization/mod.rs

```diff
@@ -0,0 +1,815 @@
+                pressurization: press,
+                engine_1: TestEngine::new(Ratio::new::<percent>(0.)),
+                engine_2: TestEngine::new(Ratio::new::<percent>(0.)),
+            }
+        }
+
+        fn set_engine_n1(&mut self, n: Ratio) {
+            self.engine_1.set_engine_n1(n);
+            self.engine_2.set_engine_n1(n)
+        }
+    }
+
+    impl Aircraft for TestAircraft {
+        fn update_after_power_distribution(&mut self, context: &UpdateContext) {
+            self.pressurization
+                .update(context, [&self.engine_1, &self.engine_2]);
+        }
+    }
+
+    impl SimulationElement for TestAircraft {
+        fn accept<V: SimulationElementVisitor>(&mut self, visitor: &mut V) {
+            visitor.visit(self);
+        }
+    }
+
+    #[test]
+    fn conversion_from_pressure_to_altitude_works() {
+        let mut test_bed = SimulationTestBed::new(|_| TestAircraft::new());
+
+        test_bed.set_ambient_pressure(Pressure::new::<hectopascal>(250.)); //Equivalent to FL340 from tables
+        test_bed.set_on_ground(true);
+        test_bed.run();
+        test_bed.run_with_delta(Duration::from_secs(20));
+
+        assert!(
+            (test_bed.query(|a| a.pressurization.cpc[0].cabin_altitude())
+                - Length::new::<foot>(34000.))
+            .abs()
+                < Length::new::<foot>(10.)
+        );
+    }
+
+    #[test]
+    fn positive_cabin_vs_reduces_cabin_pressure() {
+        let mut test_bed = SimulationTestBed::new(|_| TestAircraft::new());
+
+        test_bed.set_indicated_airspeed(Velocity::new::<knot>(101.));
+        test_bed.set_vertical_speed(Velocity::new::<foot_per_minute>(1000.));
+
+        test_bed.run_with_delta(Duration::from_secs(10));
+        let last_cabin_pressure = test_bed.query(|a| a.pressurization.cpc[0].cabin_pressure());
+        test_bed.run_with_delta(Duration::from_secs(10));
+        assert!(last_cabin_pressure > test_bed.query(|a| a.pressurization.cpc[0].cabin_pressure()));
+    }
+
+    #[test]
+    fn seventy_seconds_after_landing_cpc_switches() {
+        let mut test_bed = SimulationTestBed::new(|_| TestAircraft::new());
+
+        test_bed.run();
+        test_bed.run_with_delta(Duration::from_secs_f64(31.));
+
+        assert!(test_bed.query(|a| a.pressurization.active_system == 1));
+
+        test_bed.set_vertical_speed(Velocity::new::<foot_per_minute>(-260.));
+        test_bed.run_with_delta(Duration::from_secs_f64(31.));
+
+        assert_eq!(
+            test_bed.query(|a| a.pressurization.cpc[0].pressure_schedule()),
+            PressureSchedule::DescentInternal
+        );
+
+        test_bed.set_indicated_airspeed(Velocity::new::<knot>(99.));
+        test_bed.set_on_ground(true);
+        test_bed.run();
+
+        assert_eq!(
+            test_bed.query(|a| a.pressurization.cpc[0].pressure_schedule()),
+            PressureSchedule::Ground
+        );
+
+        test_bed.run_with_delta(Duration::from_secs_f64(90.));
```

Comment:
**issue**: test name is `seventy_seconds_after_landing_cpc_switches`, but tests 90 seconds.

Reply:
Yes that's my bad! I tested this to make sure that it changed after 70 seconds and it _stayed_ on the same CPC after that. I've added a couple more asserts in the test now to make it more accurate to what it should be testing.